From e095de96bca04508beb31d8802a6177c5c34df4d Mon Sep 17 00:00:00 2001
From: "FeRD (Frank Dana)" <ferdnyc@gmail.com>
Date: Wed, 14 Nov 2018 17:04:17 -0500
Subject: [PATCH] Pass remaining command line args to Qt

Make sure we don't error out when a project file name is passed on the
command line.

Fixes #1872
---
 src/launch.py | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/launch.py b/src/launch.py
index 76c5d5e9..a7f676f7 100755
--- a/src/launch.py
+++ b/src/launch.py
@@ -41,7 +41,7 @@
  """
 
 import sys
-from argparse import ArgumentParser
+from argparse import ArgumentParser, REMAINDER
 
 try:
     from classes import info
@@ -67,6 +67,7 @@ def main():
                         action='store_true', help='List all language '
                         'codes supported by OpenShot')
     parser.add_argument('-V', '--version', action='store_true')
+    parser.add_argument('remain', nargs=REMAINDER)
 
     args = parser.parse_args()
 
@@ -94,8 +95,11 @@ def main():
     log.info("   OpenShot (version %s)" % info.SETUP['version'])
     log.info("------------------------------------------------")
 
-    # Create Qt application
-    app = OpenShotApp(sys.argv)
+    # Create Qt application, pass any unprocessed arguments
+    argv = [sys.argv[0]]
+    for arg in args.remain:
+        argv.append(arg)
+    app = OpenShotApp(argv)
 
     # Run and return result
     sys.exit(app.run())
-- 
2.19.1

